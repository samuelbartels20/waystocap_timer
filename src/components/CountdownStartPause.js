import React from 'react';
import { Button, Icon } from 'semantic-ui-react';
import PropTypes from 'prop-types';

function CountdownStartPause({ disabled, isRunning, onClick }) {
  const icon = isRunning ? 'pause' : 'play';
  return (
    <Button disabled={disabled} onClick={onClick} icon>
      <Icon name={icon} />
    </Button>
  );
}

CountdownStartPause.propTypes = {
  disabled: PropTypes.bool,
  isRunning: PropTypes.bool.isRequired,
  onClick: PropTypes.func,
};

export default CountdownStartPause;
