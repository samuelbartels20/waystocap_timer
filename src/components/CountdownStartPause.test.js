import React from 'react';
import CountdownStartPause from './CountdownStartPause';
import { shallow } from 'enzyme';

it('shows a different icon for running and not running', () => {
  const notRunning = shallow(<CountdownStartPause isRunning={false} />);
  expect(notRunning).toMatchSnapshot();
  const running = shallow(<CountdownStartPause isRunning={true} />);
  expect(running).toMatchSnapshot();
});
