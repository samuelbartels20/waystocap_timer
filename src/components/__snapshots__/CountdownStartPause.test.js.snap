// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows a different icon for running and not running 1`] = `ShallowWrapper {}`;

exports[`shows a different icon for running and not running 2`] = `ShallowWrapper {}`;
